packages/react/src/primitives/Autocomplete/AutocompleteMenu.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+
+import { HighlightMatch } from '../HighlightMatch/HighlightMatch';
+import { Loader } from '../Loader';
+import { AutocompleteOption } from './AutocompleteOption';
+import { ScrollView } from '../ScrollView';
+import { View } from '../View';
+import { ComponentClassNames } from '../shared/constants';
+import { ComponentText } from '../shared/constants';
+import { isFunction } from '../shared/utils';
+import type { Primitive, AutocompleteMenuProps, Option } from '../types';
+
+export const AutocompleteMenu: Primitive<AutocompleteMenuProps, 'div'> = ({
+  activeIdx,
+  activeOptionId,
+  ariaLabel,
+  Header = null,
+  Footer = null,
+  Loading = null,
+  Empty = null,
+  isControlled,
+  isCustomFiltering = false,
+  isLoading,
+  isOpen,
+  listboxId,
+  onSelect,
+  optionBaseId,
+  options = [],
+  renderOption,
+  setActiveIdx,
+  setIsOpen,
+  setValue,
+  value,
+  ...rest
+}) => {
+  const MenuHeader = React.useCallback(() => {
+    return (
+      Header && (
+        <View className={ComponentClassNames.AutocompleteMenuHeader}>
+          {Header}
+        </View>
+      )
+    );
+  }, [Header]);
+
+  const MenuFooter = React.useCallback(() => {
+    return (
+      Footer && (
+        <View className={ComponentClassNames.AutocompleteMenuFooter}>
+          {Footer}
+        </View>
+      )
+    );
+  }, [Footer]);
+
+  const MenuLoading = React.useCallback(
+    () =>
+      Loading ? (
+        <View className={ComponentClassNames.AutocompleteMenuLoading}>
+          {Loading}
+        </View>
+      ) : (
+        <View className={ComponentClassNames.AutocompleteMenuLoading}>
+          <Loader />
+          {ComponentText.Autocomplete.loadingText}
+        </View>
+      ),
+    [Loading]
+  );
```

Comment:
Would consider refactoring this to make it a bit simpler to read and more straightforward, `Loading ?` reads as though we are using a `Loading` state to determine the value of `MenuLoading`:

```
const MenuLoading = CustomLoader ?? (
  <>
    <Loader />
    {ComponentText.Autocomplete.loadingText}
  </>
);

// in the return statement
<View className={ComponentClassNames.AutocompleteMenuLoading}>
  <MenuLoading />
</View>
```

Reply:
As a follow up here, we could also consider updating the type of `Loading` to expect `children` so we could just declare a default loading component outside of `AutocompleteMenu`:
```
// generic for custom props the user may want to have on their `Loading` component
type Loading<P = {}> = React.ComponentType<{ children: ReactNode; } & P>;

// in AutocompleteMenu.tsx
const DefaultLoading = ({ children ]) => {
  <>
    <Loader />
    {ComponentText.Autocomplete.loadingText}
  </>
}

const AutocompleteMenu = () => {
 const LoadingComponent = Loading ?? DefaultLoading; 

  return (
    <View className={ComponentClassNames.AutocompleteMenuLoading}>
      <LoadingComponent>{ComponentText.Autocomplete.loadingText}</LoadingComponent>
    </View>
  )
}
```

Reply:
I would prefer the first idea